src/v/kafka/server/handlers/fetch.cc

```diff
@@ -241,22 +287,24 @@ static ss::future<> do_fill_fetch_responses(
         }
         return std::move(*res.reader)
           .consume(kafka_batch_serializer(), model::no_timeout)
-          .then(
-            [so = res.start_offset,
-             hw = res.high_watermark,
-             lso = res.last_stable_offset,
-             pid = res.partition,
-             resp_it = resp_it](kafka_batch_serializer::result res) mutable {
-                fetch_response::partition_response resp{
-                  .partition_index = pid,
-                  .error_code = error_code::none,
-                  .records = batch_reader(std::move(res.data)),
-                };
-                resp_it.set(std::move(resp));
-                resp_it->partition_response->log_start_offset = so;
-                resp_it->partition_response->high_watermark = hw;
-                resp_it->partition_response->last_stable_offset = lso;
-            })
+          .then([so = res.start_offset,
+                 hw = res.high_watermark,
+                 lso = res.last_stable_offset,
+                 pid = res.partition,
+                 rres = &res,
```

Comment:
nit: I think you can do `&rres = res` to capture the reference rather than a raw pointer. alternatively, capture `aborted{std::move(res.aborted_transactions}}`